frontend/app-development/features/dataModelling/utils/validationUtils.ts

```diff
@@ -39,11 +40,20 @@ export const findFileNameError = (
 };
 
 const isNameFormatValid = (fileNameWithoutExtension: string): boolean => {
-  const fileNameRegex: RegExp = /^[a-zA-Z][a-zA-Z0-9_.\-æÆøØåÅ ]*$/;
-  return Boolean(fileNameWithoutExtension.match(fileNameRegex));
+  return Boolean(fileNameWithoutExtension.match(DATA_MODEL_NAME_REGEX));
 };
 
 const doesFileExistInMetadata = (
   appMetadata: ApplicationMetadata,
   fileNameWithoutExtension: string,
 ): boolean => appMetadata.dataTypes?.some((dataType) => dataType.id === fileNameWithoutExtension);
+
+export const extractDataTypeNamesFromAppMetadata = (
+  appMetadata?: ApplicationMetadata,
+): string[] => {
+  if (appMetadata?.dataTypes) {
+    return appMetadata.dataTypes.map((dataType) => dataType.id);
+  } else {
+    return [];
+  }
```

Comment:
```suggestion
return (
    appMetadata?.dataTypes
      ?.filter((dataType) => dataType.appLogic?.classRef === undefined)
      .map((dataType) => dataType.id) || []
  );
```

Reply:
Hva er egentlig `classRef` her? 🤔

Reply:
Hm, ja når du sier det så blir dette kanskje litt feil i dette caset når man ser på funksjonen isolert. ClassRef er et felt som kun er på de  dataTypene som er knyttet til en datamodell. Men denne funksjonen ville vel finne alle dataTypeIder (Selvom der den brukes så ser den kun ut til å bry seg om de som _ikke_ er datamodeller, om jeg ikke tar feil? 🧐)

Reply:
Ja, det stemmer 😊 Det var det valideringen sjekket før jeg tok fatt i oppgaven.

Reply:
Da er det litt opp til deg om du vil tilpasse navnet og i tillegg filtrere på ikke-classRef, eller beholde den slik den er. 
Evt kan den forenkles til dette da 😊 
```
return (
    appMetadata?.dataType.map((dataType) => dataType.id) || []
  );
```

Jeg tror at å filtrere ut de som har classRef gjør at valideringskoden blir mer logisk. For da har det egt ikke noe å si hvilken rekkefølge if-ene står i. Men i dag må vi sjekke datamodeller først for at returteksten i denne sjekken er riktig. Om du skjønner 🤔

Reply:
Jeg går for forenklingen din over 😊